dist/jose-jwe.js

```diff
@@ -553,10 +562,10 @@ JoseJWE.prototype.encryptCek = function(key_promise, cek_promise) {
  */
 JoseJWE.prototype.encryptPlainText = function(cek_promise, plain_text) {
   // Create header
-  var jwe_protected_header = Utils.Base64Url.encode(JSON.stringify({
-    "alg": this.key_encryption.jwe_name,
-    "enc": this.content_encryption.jwe_name
-  }));
+  this.headers.alg = this.key_encryption.jwe_name;
```

Comment:
Are `JoseJWE` object supposed to be reusable? You should merge the user headers and the managed headers into a new dictionary instead of mutating the `JoseJWE` object as a side effect of encryption.